### ISSUE TITLE
When using a branch for GIT_TAG updates to the branch are not detected by FetchDependency.

### ISSUE THREAD

**Issue by WashuHakubi** [jpetrie/fetch-dependency#4]
When using a branch for GIT_TAG updates to the branch are not detected by FetchDependency.

This is because the version.txt file stores the branch name and not the SHA of the branch.

**Comment by jpetrie** [comment]
@WashuHakubi In b5bcfb2b6dc86ab12bdd74355966e99d913c6efb I removed `GIT_TAG` from the content I write to `version.txt`, which means the logic related to pulling from the source repository should exactly match `ExternalProject_Add`'s. I think this fixes this issue, as I was able to rerun CMake and see it pull in changes to a test repository that I made. I did make sure to `GIT_TAG origin/main` as recommended by `ExternalProject_Add`'s documentation.
